test/unit/test_gateway.c

```diff
@@ -1080,6 +1080,114 @@ TEST_CASE(exec, undo_not_leader_pending_other_elected, NULL)
 	return MUNIT_OK;
 }
 
+/* Succesfully vacuum a database */
+TEST_CASE(exec, vacuum, NULL)
+{
+	struct exec_fixture *f = data;
+	uint64_t stmt_id;
+	(void)params;
+	CLUSTER_ELECT(0);
+
+	/* Create some free pages */
+	LOWER_CACHE_SIZE;
+	EXEC("CREATE TABLE test (n INTEGER PRIMARY KEY NOT NULL)");
+	EXEC("WITH RECURSIVE seq(n) AS ("
+		"    SELECT 1 UNION ALL     "
+		"    SELECT n+1 FROM seq    "
+		"    WHERE  n < 10000       "
+		")                          "
+		"INSERT INTO test(n)        "
+		"SELECT n FROM seq          "
+	)
+	EXEC("DELETE FROM test WHERE n <= 5000");
+
+	PREPARE("VACUUM");
```

Comment:
Is it worth checking that VACUUM actually did something? 

The following queries might give us the picture of changes.
```
sqlite> PRAGMA freelist_count;
sqlite> PRAGMA page_count;
sqlite> PRAGMA page_size;
```

Reply:
Yes, makes sense.